### PR TITLE
Completed calendar feature implementation documentation

### PR DESCRIPTION
**Pull request by chenxing1992**
#49

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -385,7 +385,55 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
 
 Certain properties of the application can be controlled (e.g App name, logging level) through the configuration file (default: `config.json`).
 
-=== [Proposed]
+=== [Proposed] Calendar Viewer Feature
+==== Current Implementation
+The Calendar Viewer mechanism is facilitated by `Google Calendar API` and reside in the `ModelManager`.
+It supports viewing/add/editing/deleting capability that modifies the state of the NUSCouples App.
+Firstly, it uses Outh 2.0 endpoints to allow users to share specific data with the application while keeping their usernames, passwords, and other information private. For example, an application can use OAuth 2.0 to obtain permission from users to store files in their Google Drives which sync to the calendar.
```
Comment (marlenekoh) [inline docs/DeveloperGuide.adoc]: typo: OAuth 2.0

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -385,7 +385,55 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
...
+The only downside to this implementation is that the user needs to connect to the internet because NUSCouples app needs to open the system browser and supply a local redirect URI to handle responses from Google's authorization server.
```
Comment (marlenekoh) [inline docs/DeveloperGuide.adoc]: Maybe can put in design considerations (alternative w & w/o internet)? or don't call it a downside cause we supposed to +++ our product, instead mention in NFR that our app requires internet.

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -385,7 +385,55 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
...
+There are several ways to make this request, and they vary based on the type of application you are building. For example, a JavaScript application might request an access token using a browser redirect to Google, while an application installed on a device that has no browser uses web service requests.
+Some requests require an authentication step where the user logs in with their Google account. After logging in, the user is asked whether they are willing to grant the permissions that your application is requesting. This process is called user consent.
+If the user grants the permission, the Google Authorization Server sends your application an access token (or an authorization code that your application can use to obtain an access token). If the user does not grant the permission, the server returns an error.
+It is generally a best practice to request scopes incrementally, at the time access is required, rather than up front. For example, an app that wants to support purchases should not request Google Wallet access until the user presses the “buy” button; see Incremental authorization.
+
```
Comment (marlenekoh) [inline docs/DeveloperGuide.adoc]: Too wordy? break up into bullet points using * or use diagram to explain to make it easier to understand. can label steps 1 - 3 in the diagram so that reader understands better

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -385,7 +385,55 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
...
+image::CalendarSequence.png[width="850"]
+
+==== Design Considerations
+===== Aspect: Implementation of view Calendar feature
+* **Alternative 1 (current choice):** Display only current month event
+** Pros: Easier to implement and Neater rather than displaying more than 1 mth.
+** Cons: Need to input cmd to filter through other month.
+* **Alternative 2:** Don't display any month until user defines
+** Pros: More interaction
```
Comment (marlenekoh) [inline docs/DeveloperGuide.adoc]: Formatting:
add full stop after all the Pros: / Cons:

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -385,7 +385,55 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
...
+** Cons: The UI will be blank at initial stage which is ugly
+
+===== Aspect: Using Open source or proprietary Calendar API
+* **Alternative 1 (current choice):** Using Google API (open source)
+** Pros: I will learnt more even i failed at the end and Google API is more versatile
```
Comment (marlenekoh) [inline docs/DeveloperGuide.adoc]: Formatting/Typos:
Open-source
Capitalise the "I"

File: docs/DeveloperGuide.adoc (hunk 2 above)
Comment (traceurgan) [reply docs/DeveloperGuide.adoc]: I do not think it need to be mentioned actually. Since our user profile is university students who are proficient in CLI and typing, i would think it is a given that they have a laptop and internet, especially since NUS has free wifi.

Comment (chenxing1992) [reply docs/DeveloperGuide.adoc]: it needs to mentioned because im writing the description and behaviour of this Auth2.0

Comment (chenxing1992) [reply docs/DeveloperGuide.adoc]: i've removed the keyword downside already

File: docs/DeveloperGuide.adoc (hunk 3 above)
Comment (chenxing1992) [reply docs/DeveloperGuide.adoc]: yes mdm
